### ISSUE TITLE
elasticsearch uses the key node, which brakes puppet-lint

### ISSUE THREAD

**Issue by oneiros-de** [rodjek/puppet-lint#563]
The elasticsearch module allows keys named `node`

```
tyntec_dev::elasticsearch2::config:
   node:
      master: false
      data: false
```

These lines breaks puppet-lint, apparently because it searches for node names:
```
/var/lib/gems/1.9.1/gems/puppet-lint-2.0.2/lib/puppet-lint/plugins/check_nodes.rb:11:in `block in check': bad value for range (ArgumentError)
        from /var/lib/gems/1.9.1/gems/puppet-lint-2.0.2/lib/puppet-lint/plugins/check_nodes.rb:6:in `each'
        from /var/lib/gems/1.9.1/gems/puppet-lint-2.0.2/lib/puppet-lint/plugins/check_nodes.rb:6:in `check'
        from /var/lib/gems/1.9.1/gems/puppet-lint-2.0.2/lib/puppet-lint/checkplugin.rb:21:in `run'
        from /var/lib/gems/1.9.1/gems/puppet-lint-2.0.2/lib/puppet-lint/checks.rb:58:in `block in run'
        from /var/lib/gems/1.9.1/gems/puppet-lint-2.0.2/lib/puppet-lint/checks.rb:56:in `each'
        from /var/lib/gems/1.9.1/gems/puppet-lint-2.0.2/lib/puppet-lint/checks.rb:56:in `run'
        from /var/lib/gems/1.9.1/gems/puppet-lint-2.0.2/lib/puppet-lint.rb:174:in `run'
        from /var/lib/gems/1.9.1/gems/puppet-lint-2.0.2/lib/puppet-lint/bin.rb:59:in `block in run'
        from /var/lib/gems/1.9.1/gems/puppet-lint-2.0.2/lib/puppet-lint/bin.rb:56:in `each'
        from /var/lib/gems/1.9.1/gems/puppet-lint-2.0.2/lib/puppet-lint/bin.rb:56:in `run'
        from /var/lib/gems/1.9.1/gems/puppet-lint-2.0.2/bin/puppet-lint:7:in `<top (required)>'
        from /usr/local/bin/puppet-lint:23:in `load'
        from /usr/local/bin/puppet-lint:23:in `<main>'

```

**Comment by rnelson0** [comment]
That looks like yaml syntax rather than a puppet file. Puppet-lint will definitely fail while parsing that. Am I reading that correctly?

**Comment by ghoneycutt** [comment]
`node` is a reserved word in puppet. If you want to use it as a key in a hash, it must be quoted.

tested by converting example to a manifest

```puppet
$config = {
  'node' => {
    master => false,
    data   => false,
  },
}
```

as @rnelson0 noted, puppet-lint validates puppet manifests, not YAML.

**Comment by oneiros-de** [comment]
Sorry for the noise.
